lib/realtime_web/controllers/channels_controller.ex

```diff
@@ -57,11 +61,12 @@ defmodule RealtimeWeb.ChannelsController do
     ],
     responses: %{
       200 => ChannelResponse.response(),
+      401 => UnauthorizedResponse.response(),
       404 => NotFoundResponse.response()
     }
   )
 
-  def show(%{assigns: %{tenant: tenant}} = conn, %{"id" => id}) do
+  def show(%{assigns: %{tenant: tenant, permissions: {:ok, %{read: true}}}} = conn, %{"id" => id}) do
```

Comment:
Should we start a type for this permissions "policy" and expand it as we go?

Reply:
good call, will create a struct for this so we can keep track and change them accordingly instead of being a magic flexible map